opentech/apply/funds/models.py

```diff
@@ -714,26 +744,73 @@ def progress_application(self, **kwargs):
         self.id = None
         self.form_fields = self.get_from_parent('get_defined_fields')(self.stage)
 
+        self.live_revision = None
+        self.draft_revision = None
         self.save()
 
         submission_in_db.next = self
         submission_in_db.save()
 
-    def save(self, *args, **kwargs):
-        for field in self.form_fields:
-            # Update the ids which are unique to use the unique name
-            if isinstance(field.block, MustIncludeFieldBlock):
-                response = self.form_data.pop(field.id, None)
-                if response:
-                    self.form_data[field.block.name] = response
+    def new_data(self, data):
+        self.is_draft = False
+        self.form_data = data
+        return self
+
+    def from_draft(self):
+        self.is_draft = True
+        self.form_data = self.draft_revision.form_data
+        return self
+
+    def create_revision(self, draft=False, force=False, by=None, **kwargs):
+        self.clean_submission()
+        current_data = ApplicationSubmission.objects.get(id=self.id).form_data
+        if current_data != self.form_data or force:
+            if self.live_revision == self.draft_revision:
+                revision = ApplicationRevision.objects.create(submission=self, form_data=self.form_data, author=by)
+            else:
+                revision = self.draft_revision
```

Comment:
if you save draft, then go back make an edit and submit, the live revision become that of the draft, not of the submitted data. Will make test and fix

Reply:
sorted. this revision needed a save.